### ISSUE TITLE
Low precision bug for batch-norm models

### ISSUE THREAD

**Issue by zoharli** [huanzhang12/alpha-beta-CROWN#1]
When I verify a CROWN-ibp pre-trained model using alpha-beta-CROWN, it consistenly shows `Result: image x prediction is incorrect, skipped.`  As a result the overall verified acc. is very low depsite CROWN-verified acc. is relatively normal.

I guess that's batch-norm support problem, because in the printed log I see two new warnings:

```
/home/zhangheng/anaconda3/envs/alpha-beta-crown/lib/python3.7/site-packages/torch/nn/functional.py:2113: TracerWarning: Converting a tensor to a Python boolean might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!

  if size_prods == 1:

/home/zhangheng/anaconda3/envs/alpha-beta-crown/lib/python3.7/site-packages/torch/onnx/symbolic_helper.py:680: UserWarning: ONNX export mode is set to inference mode, but operator batch_norm is set to training  mode. The model will be exported in inference, as specified by the export mode.

  training_mode + ", as specified by the export mode.")
```

I tried some scripts without batchnorm included in `exp_configs` folder, there are no these two warnings.

Here are the log file, please have a look, thank you!

[CROWN-ibp-model-verify.txt](https://github.com/huanzhang12/alpha-beta-CROWN/files/7586907/CROWN-ibp-model-verify.txt)

**Comment by huanzhang12** [comment]
@zoharli Thanks for reporting and I will try to reproduce this on my side and keep your posted.

**Comment by huanzhang12** [comment]
@zoharli We've found the issue for this model. The problem is within the model loader, which did not set the model into `eval()` mode, so the batchnorm mean and variance were unexpectedly changed. This is easy to fix in `utils.py`:

https://github.com/huanzhang12/alpha-beta-CROWN/blob/590412434734bf80863cf37da8421f422e39ef37/complete_verifier/utils.py#L178-L180

Just add `model_ori.eval()` after the above lines to put the model in evaluation mode. We will fix this in our repository, but before we update our repository you can simply add `model_ori.eval()` and it should solve this problem.

Clean acc. should not be lower than CROWN acc, if that still happens to you after adding `model_ori.eval()` please let me know, thanks!

**Comment by zoharli** [comment]
Oh, I get it. Thank you, Truly appreciate your timely help!
